qiskit/circuit/library/standard_gates/multi_control_rotation_gates.py

```diff
@@ -82,38 +83,52 @@ def _apply_mcu_graycode(circuit, theta, phi, lam, ctls, tgt, use_basis_gates):
         last_pattern = pattern
 
 
-def linear_depth_mcv(
+def mcsu2_real_diagonal(
     circuit,
-    x,
-    z,
+    unitary: np.ndarray,
     controls: Union[QuantumRegister, List[Qubit]],
-    target: Qubit,
+    target: Union[Qubit, int],
     ctrl_state: str = None,
 ):
     """
-    Apply multi-controlled SU(2) gate [[z*, x],[-x, z]] with x real and z complex.
+    Apply multi-controlled SU(2) gate with one real diagonal.
     https://arxiv.org/abs/2302.06377
 
     Args:
         circuit (QuantumCircuit): The QuantumCircuit object to apply the diagonal operator on.
-        x (float): real parameter for the single-qubit operators
-        z (complex): complex parameter for the single-qubit operators
+        unitary (ndarray): SU(2) unitary matrix with one real diagonal
         controls (QuantumRegister or list(Qubit)): The list of control qubits
-        target (Qubit): The target qubit
-        ctrl_state (str): control state of the operator SU(2) operator U
+        target (Qubit or int): The target qubit
+        ctrl_state (str): control state of the operator SU(2) operator
 
     Raises:
         QiskitError: parameter errors
     """
     # pylint: disable=cyclic-import
     from qiskit.circuit.library import MCXVChain
     from qiskit.extensions import UnitaryGate
+    from qiskit.quantum_info.operators.predicates import is_unitary_matrix
+
+    if not is_unitary_matrix(unitary):
+        raise QiskitError("parameter unitary in linear_depth_mcv must be an unitary matrix")
+
+    if unitary.shape != (2, 2):
+        raise QiskitError("parameter unitary in linear_depth_mcv must be a 2x2 matrix")
+
+    is_main_diag_real = np.isclose(unitary[0, 0].imag, 0.0) and np.isclose(unitary[1, 1].imag, 0.0)
```

Comment:
These could also be `np.isreal(v.diagonal()).all()` and `np.isreal(np.fliplr(v).diagonal()).all()`

Reply:
`np.isreal` has different behavior. For instance, `np.isclose((1+1e-17j).imag, 0.)` returns `True` and `np.isreal(1+1e-17j)` returns `False`.

Reply:
Good point. I can think of a "fix" but at that point it may not be much shorter anyways.

Reply:
I notice you made the change  but I'm ok if you want to revert it

Reply:
I reverted the change.

Reply:
I merged main locally and push. Then I did a force push to clear the PR history.